engine/vcs/convert_test.go

```diff
@@ -0,0 +1,274 @@
+package vcs
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tonistiigi/fsutil/types"
+)
+
+func TestConvertRef(t *testing.T) {
+	bkClientDirFalse := &MockBuildkitClient{
+		StatFunc: func(ctx context.Context, path string, followLinks bool) (*types.Stat, error) {
+			return &types.Stat{Mode: uint32(os.ModeDevice)}, nil // stat.Dir() returns false
+		},
+	}
+
+	testCases := []struct {
+		name           string
+		urlStr         string
+		expectedUrlStr string
+	}{
+		// SSH ref without version or subdir
+		{
+			name:           "SSH ref without .git",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse",
+		},
+		{
+			name:           "SSH ref with .git",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse.git",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse.git",
+		},
+
+		// SSH ref with version
+		{
+			name:           "SSH ref with version",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse#v0.9.1",
+		},
+		{
+			name:           "SSH ref with .git and version",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse.git@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse.git#v0.9.1",
+		},
+
+		// SSH ref with version and subdir
+		{
+			name:           "SSH ref with subdir and version",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse/subdir1/subdir2@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse#v0.9.1:subdir1/subdir2",
+		},
+		{
+			name:           "SSH ref with .git, subdir and version",
+			urlStr:         "ssh://git@github.com/shykes/daggerverse.git/subdir1/subdir2@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com/shykes/daggerverse.git#v0.9.1:subdir1/subdir2",
+		},
+
+		// Implicit SSH ref without version or subdir
+		{
+			name:           "Git ref without .git",
+			urlStr:         "git@github.com:sipsma/daggerverse",
+			expectedUrlStr: "ssh://git@github.com:sipsma/daggerverse",
+		},
+		{
+			name:           "Git ref with .git",
+			urlStr:         "git@github.com:sipsma/daggerverse.git",
+			expectedUrlStr: "ssh://git@github.com:sipsma/daggerverse.git",
+		},
+
+		// Implicit SSH ref with version
+		{
+			name:           "Git ref with version",
+			urlStr:         "git@github.com:sipsma/daggerverse@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com:sipsma/daggerverse#v0.9.1",
+		},
+		{
+			name:           "Git ref with .git and version",
+			urlStr:         "git@github.com:sipsma/daggerverse.git@v0.9.1",
+			expectedUrlStr: "ssh://git@github.com:sipsma/daggerverse.git#v0.9.1",
+		},
```

Comment:
Hm, this shouldn't be a valid syntax.

The supported syntaxes should either be:
- The older buildkit ref style
- The newer module ref style

I don't think this is either of the above?

Reply:
Ok 👌 
I wanted to cover all the test cases of the current TestParseGit: https://github.com/dagger/dagger/blob/main/cmd/dagger/module_test.go#L148

Reply:
Oh agreed, it's just that this specific case isn't in there.